### ISSUE TITLE
Docker image (vg v1.21.0) does ```ls -al``` and exit

### ISSUE THREAD

**Issue by ManuelTgn** [vgteam/vg#2634]
**1. What were you trying to do?**
I was trying to run the docker image for vg v1.21.0

**3. What actually happened?**
After pulling the container and the typing 
```
docker run -it quay.io/vgteam/vg:v1.21.0
```
instead of running it seems to do ```ls -al``` and then the image is closed. 

**5. What data and command can the vg dev team use to make the problem happen?**
```
docker pull quay.io/vgteam/vg:v1.21.0
```
```
docker run -it quay.io/vgteam/vg:v1.21.0
```

Thanks for helping

**Comment by glennhickey** [comment]
Doesn't look like there's an entry point, so you need to specify the command you want to run at the end of your command line.  add `vg` to run vg or `bash` for a shell if that's what you're looking for.  

```
docker run --rm quay.io/vgteam/vg:v1.21.0 vg
vg: variation graph tool, version v1.21.0 "Fanano"

usage: vg <command> [options]

main mapping and calling pipeline:
  -- construct     graph construction
  -- index         index graphs or alignments for random access or mapping
  -- map           MEM-based read alignment
  -- augment       augment a graph from an alignment
  -- pack          convert alignments to a compact coverage index
  -- call          call or genotype VCF variants
  -- help          show all subcommands

For more commands, type `vg help`.
For technical support, please visit: https://www.biostars.org/t/vg/

```

**Comment by ManuelTgn** [comment]
Specifying a command there are no problems, but I was looking to open the image in interactive mode with ```-it```, as it was possible with v1.20.0 and older versions.                                                      I'm asking this because I'm creating a docker image that contains vg image.

**Comment by adamnovak** [comment]
Yeah, it definitely is using an `ls` as the image's default command and I have no idea why.

A workaround for your use case @ManuelTgn is to run `docker run -it quay.io/vgteam/vg:v1.21.0 /bin/bash` which will put you inside the image interactively by explicitly invoking a shell.

The default command isn't actually set in our Dockerfile, but we do run a bunch of `ls` commands, so maybe in the absence of a real command set on the image it just picks up a command from the build process somehow?

I think the default command ought to be just `vg`, though, in which case running `docker run -it quay.io/vgteam/vg:v1.21.0` would just print vg's help, and still not give you a shell.

**Comment by adamnovak** [comment]
OK, the problem is that somehow, despite the lack of any `CMD` in the Dockerfile, a non-default `Cmd` is being set in the container `Config`:

```
docker inspect --format='{{.Config.Cmd}}' quay.io/vgteam/vg:v1.21.0
[/bin/sh -c ls -lah /vg || echo "No vg directory exists yet"]
docker inspect --format='{{.Config.Cmd}}' quay.io/vgteam/vg:v1.20.0
[/bin/bash]
```

This command is coming from one of several instances we have in the Dockerfile. In particular it is the last command run in the base image that we use to build both the build and (final) run images.

https://github.com/vgteam/vg/blob/4d48086a7ab6362ec911e318278722add05c73c0/Dockerfile#L9

Dockerfiles aren't *supposed* to work like that, and I just did a build myself that didn't produce this weird command, so I suspect this might be some kind of Quay.io build system bug. I suspect we could work around it by specifying our own `CMD` in the `Dockerfile`.

**Comment by adamnovak** [comment]
I've kicked off a new Quay build at https://quay.io/repository/vgteam/vg/build/ff01a062-cb51-4d6c-b70b-5f40e52ec837 (the page for which weirdly loads fine and *then* claims you need to log in to Quay if you aren't logged in), and when that is done we can look at the v1.21.0 tag again and see if it still weirdly has this command.

**Comment by faithokamoto** [comment]
Seems this was handled.
